hs_core/views/__init__.py

```diff
@@ -602,7 +602,7 @@ def publish(request, shortkey, *args, **kwargs):
 def set_resource_flag(request, shortkey, *args, **kwargs):
     # only resource owners are allowed to change resource flags
     res, _, user = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.SET_RESOURCE_FLAG)
-    t = request.POST.get('t', request.data.get('t'))
+    t = request.POST.get('t', request.data.get('t', None))
```

Comment:
@aphelionz  I did some testing which conformed that ```request.data``` will always be evaluated. That means when this function is used by the UI, it will raise AttributeError since the ```request``` object wont have the ```data``` attribute. This is the issue I have found in testing of 1.9.6 on beta as it relates to zipping folder or moving/renaming files or folders.